### PR TITLE
Add CoilImagesPlugin Module

### LINKED ISSUE #166: Feature request: Support Coil image loader
[Coil](https://github.com/coil-kt/coil) is a new and popular image loading library for android. It would be great if Markwon offered a first party artifact, though it may be more difficult since Coil is a Kotlin-first API

### PR DESCRIPTION
**Pull request by tylerbwong**
Closes #166 

This adds support for loading images using Coil through a new `markwon-image-coil` module based off the implementations of the existing image plugins. I'm not too familiar with any gotchas there would be with interactions between the Markwon and Coil APIs so any guidance would be much appreciated!

### CODE REVIEW

File: docs/docs/v4/image-coil/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Image Coil
```
Comment (noties) [inline docs/docs/v4/image-coil/README.md]: 😍 ! Very nice touch to include documentation!

File: docs/docs/v4/image-coil/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+<MavenBadge4 :artifact="'image-coil'" />
+
+Image loading based on `Coil` library
+
+```kt
```
Comment (noties) [inline docs/docs/v4/image-coil/README.md]: Unfortunately syntax highlighting won't work for `kt`, please change to `kotlin`

File: markwon-image-coil/src/main/java/io/noties/markwon/image/coil/CoilImagesPlugin.java [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+import io.noties.markwon.MarkwonConfiguration;
+import io.noties.markwon.MarkwonSpansFactory;
+import io.noties.markwon.image.AsyncDrawable;
+import io.noties.markwon.image.AsyncDrawableLoader;
+import io.noties.markwon.image.AsyncDrawableScheduler;
+import io.noties.markwon.image.DrawableUtils;
+import io.noties.markwon.image.ImageSpanFactory;
+
+/**
+ * @since 4.0.0
```
Comment (noties) [inline markwon-image-coil/src/main/java/io/noties/markwon/image/coil/CoilImagesPlugin.java]: You can put here some attribution if you wish, like your name and email, but it's optional

File: docs/docs/v4/image-coil/README.md (hunk 2 above)
Comment (sridhar-vadlamani-ck) [inline docs/docs/v4/image-coil/README.md]:
```suggestion
```kotlin
```

**Pull request merged**